pkg/kubelet/kubelet_pods_test.go

```diff
@@ -292,13 +353,27 @@ fe00::1	ip6-allnodes
 fe00::2	ip6-allrouters
 203.0.113.1	podFoo.domainFoo	podFoo
 fd00::6	podFoo.domainFoo	podFoo
+`,
+		},
+		{
+			hostIPs:        []string{"203.0.113.1", "fd00::6"},
+			hostName:       "podFoo",
+			hostDomainName: "domainFoo",
+			hostAliases:    []v1.HostAlias{},
+			isSupportIPv6:  false,
+			expectedContent: `# Kubernetes-managed hosts file.
+127.0.0.1	localhost
+203.0.113.1	podFoo.domainFoo	podFoo
+fd00::6	podFoo.domainFoo	podFoo
 `,
 		},
 	}
 
 	for _, testCase := range testCases {
 		actualContent := managedHostsFileContent(testCase.hostIPs, testCase.hostName, testCase.hostDomainName, testCase.hostAliases)
-		assert.Equal(t, testCase.expectedContent, string(actualContent), "hosts file content not expected")
+		if testCase.isSupportIPv6 == isSupportIPv6() {
+			assert.Equal(t, testCase.expectedContent, string(actualContent), "hosts file content not expected")
+		}
```

Comment:
this is skipping use cases, the tests can not depend on the host sysctl, you have to inject that dependency.
One option is make `isSupportIPv6` a boolean that you pass as a parameter and get the value of the sysctl in 
```
		// if Pod is not using host network, create a managed hosts file with Pod IP and other information.
		hostsFileContent = managedHostsFileContent(hostIPs, hostName, hostDomainName, hostAliases, isIPv6)
```